### PR TITLE
Display code string

### LINKED ISSUE #1604: Autocomplete chip is empty if display text not specified
**Describe the bug**
I have an answerOption with a bunch of codes without display text, e.g.

```json
"answerOption":
[
  {
    "valueCoding":
    {
      "code": "LA11231-0",
      "system": "http://loinc.org"
    }
  },
  {
    "valueCoding":
    {
      "code": "LA24366-9",
      "system": "http://loinc.org"
    }
  },
...
```

When using autocomplete, it searches on all the `code` text which I believe is reasonable, but when I select one the chip itself is empty.

**Component**
SDC Library

**To Reproduce**
Load a questionnaire with answerOption items without `display` values.

**Expected behavior**
The `code` is used to fill the chip text if `display` is missing, since that's what was shown while doing the autocomplete.

**Screenshots**

https://user-images.githubusercontent.com/7772901/189761145-9ad52a3a-2701-43e5-888e-c0ab6dfce546.mp4

### PR DESCRIPTION
**Pull request by santosh-pingle**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes #1604

**Description**
Updated code string if display string is not present.

**Alternative(s) considered**
Have you considered any alternatives? And if so, why have you chosen the approach in this PR?

**Type**
Choose one: Bug fix 
**Screenshots (if applicable)**

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/CODE_OF_CONDUCT.md).
- [x] I have read the [Contributing](https://github.com/google/android-fhir/wiki/Contributing) page.
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate ).
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach.
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project.
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally.
- [x] I have built and run the demo app(s) to verify my change fixes the issue and/or does not break the demo app(s).

### CODE REVIEW

File: datacapture/src/main/java/com/google/android/fhir/datacapture/views/factories/AutoCompleteViewHolderFactory.kt [hunk 1]
```diff
@@ -141,7 +141,12 @@ internal object AutoCompleteViewHolderFactory :
 
         val chip = Chip(chipContainer.context, null, R.attr.questionnaireChipStyle)
         chip.id = View.generateViewId()
-        chip.text = answer.valueCoding.display
+        chip.text =
+          if (answer.valueCoding.display.isNullOrBlank()) {
+            answer.valueCoding.code
+          } else {
+            answer.valueCoding.display
+          }
```
Comment (jingtang10) [inline datacapture/src/main/java/com/google/android/fhir/datacapture/views/factories/AutoCompleteViewHolderFactory.kt:149]:
```suggestion
          answer.valueCoding.let {
            if (it.display.isNullOrBlank()) {
              it.code
            } else {
              it.display
            }
          }
```

File: datacapture/src/main/java/com/google/android/fhir/datacapture/views/factories/AutoCompleteViewHolderFactory.kt [hunk 2]
```diff
@@ -172,7 +177,12 @@ internal object AutoCompleteViewHolderFactory :
           addNewChipIfNotPresent(answer)
         } else {
           (chipContainer[0] as Chip).apply {
-            text = answer.valueCoding.display
+            text =
```
Comment (jingtang10) [inline datacapture/src/main/java/com/google/android/fhir/datacapture/views/factories/AutoCompleteViewHolderFactory.kt:180]: same here

Comment (jingtang10) [reply datacapture/src/main/java/com/google/android/fhir/datacapture/views/factories/AutoCompleteViewHolderFactory.kt:180]: you can also make a private extension function/property called Coding.displayOrCode

```
private fun Coding.displayOrCode = ...
```

**Review by jingtang10 — APPROVED** [review]
thanks @santosh-pingle

File: datacapture/src/main/java/com/google/android/fhir/datacapture/views/factories/AutoCompleteViewHolderFactory.kt (hunk 2 above)
Comment (santosh-pingle) [reply datacapture/src/main/java/com/google/android/fhir/datacapture/views/factories/AutoCompleteViewHolderFactory.kt]: done.

File: datacapture/src/main/java/com/google/android/fhir/datacapture/views/factories/AutoCompleteViewHolderFactory.kt (hunk 1 above)
Comment (santosh-pingle) [reply datacapture/src/main/java/com/google/android/fhir/datacapture/views/factories/AutoCompleteViewHolderFactory.kt]: done.

**Review by jingtang10 — APPROVED** [review]

**Pull request merged**